system/src/system_network_internal.h

```diff
@@ -622,6 +624,24 @@ class ManagedNetworkInterface : public NetworkInterface
         }
     }
 
+    bool isOn() override
+    {
+#if PLATFORM_ID == PLATFORM_ELECTRON // Electron
+        return !WLAN_INITIALIZED || SPARK_WLAN_STARTED;
```

Comment:
There probably shouldn't be a `PLATFORM_ID == PLATFORM_ELECTRON` check and the condition for on is as Brett says: `SPARK_WLAN_STARTED && WLAN_INITIALIZED`.

Reply:
We need to define what `isOn()`/`isOff()` means. Since the modem may reside in a state that it is powered on, but not initialized. In corresponding to Gen3 platforms, the NCP power state may be on, while the NCP client is not ready to accept AT commands. 

The implementation of the new APIs on Gen3 does return the modem power state, despite of whether the NCP client is ready or not. 

We had a fix for Electron power leakage: https://github.com/particle-iot/device-os/pull/2197, which means that the SARA module is on by default on device powering up. And I just did a test to verify that the BCM module is on by default as well:
```
void setup() {
    pinMode(D0, INPUT_PULLUP);
    if (digitalRead(D0) == HIGH) {
        Particle.connect();
        waitUntil(Particle.connected);
    } else {
        delay(20s);
        WiFi.on();
        delay(20s);
        WiFi.off();
    }
}
```
The power consumption on powering up is higher than after we turn off the module on Photon.

If `isOn()` intends to indicate the power state, than we should return `return !WLAN_INITIALIZED || SPARK_WLAN_STARTED;` for all Gen2 platforms here.

Reply:
If you think from the application perspective you care about two things:
1. Is whatever I need to be off and not consume power really off or do I need to perform some action to turn it off (both logically and physically off)
2. Is whatever I turned on really on and ready to proceed further (in our case - accept commands or initiate connection for example) (both physically and logically on - i.e. passed some low-level initialization)

That would at least be my interpretation.

Reply:
If I understand correctly, the modem may reside in one of the following states:
**Physically Off <-> Physically On but Logically off <-> Logically On**
So,
1. The `isOff()` should return `true` only if the modem resides in the **Physically Off** state
2. The `isOn()` should return `true` only if the modem resides in the **Logically On** state.

Reply:
1. Physically off and logically off (which means that it's not supposed to get turned on at all). For example we encountered some problem and had to hard reset modem: those conditions should not report `isOff()` as `true`.
2. Correct.

Reply:
That makes sense 👍